engine/crates/parser-sdl/src/schema_coord.rs

```diff
@@ -0,0 +1,47 @@
+/// Helper enum for specifying the location of errors
+#[derive(Clone, Copy, Debug)]
+pub enum SchemaCoord<'a> {
+    Field(&'a str, &'a str),
+    Entity(&'a str, &'a str),
+}
+
+impl std::fmt::Display for SchemaCoord<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            SchemaCoord::Field(ty, field) => {
+                write!(f, "{ty}.{field}")
+            }
+            SchemaCoord::Entity(ty, key) => {
+                write!(f, "federation key `{key}` on the type {ty}")
+            }
+        }
+    }
+}
+
+impl SchemaCoord<'_> {
+    pub fn into_owned(self) -> OwnedSchemaCoord {
```

Comment:
an alternative could be using `Cow` to store the value and return a `'static` with `into_owned()`

Reply:
I almost went down that route, but it makes constructing them a bit more annoying so I took this approach instead